### ISSUE TITLE
Usability features

### ISSUE THREAD

**Issue by es20490446e** [dosbox-staging/dosbox-staging#845]
I have developed a wrapper around Dosbox and Wine, called [exeCute](https://gitlab.com/es20490446e/exeCute), that smooths some usability problems that I found on them. Mainly:

- If to launch an exe (or bat) should be done on Wine or Dosbox, as Dosbox usually behaves weirdly when launched from Wine.
- Automounting disk images found within the launched program folder, or if no image is found auto-mounting that folder as  'D' for maximum compatibility.
- Recognizing '.gog' as a disk image extension.
- If multiple images are mounted, display a notification telling a shortcut for changing among them.
- When launching on Wine, as it takes a while to launch a program, displaying a notification that the program is being launched.
- Not interpreting "Alt+F4" as a program failure.

In case you decided to implement some of them within Dosbox. Keep up the good work! 👍

**Comment by dreamer** [comment]
Yeah, WineVDM works fine, but it invokes dosbox in a way that's very specific to Wine (e.g. it makes sure that Wine's virtual C: drive is the same as DOSBox virtual C: drive) - this works great for unpacked/installed software (intended usecase), but won't work well if user wants to imgmount some CD images (that's out of Wine's scope).

Overall this looks very interesting - thanks @es20490446e!

> Not interpreting "Alt+F4" as a program failure.

@kcgen fixed this few weeks ago :) (but only on master branch - will be released in 0.77.0)

**Comment by es20490446e** [comment]
Thank you 👍

If you let me know when you make the next release, I would remove the dirty work-around I made of Dosbox always succeeding.

**Comment by FeignInt** [comment]
> Thank you +1
> 
> If you let me know when you make the next release, I would remove the dirty work-around I made of Dosbox always succeeding.

For some degree of version compatibility you could add some condition on the version

for example:

```shell
readarray -t -d. dosbox_version < <(
  grep -m1 -Po "0\.\d{2}(\.\d+)?" < <( dosbox --version )
)
```
Gives you an array (with crude SVN vs Staging detection)

You can then use that array to enable/disable "work-arounds" or simply prompt user to upgrade their DOSBox Staging

---

couple of hints,

I find [this bash guide](https://mywiki.wooledge.org/BashGuide) excellent, there is a newer version linked via that but I haven't had much chance to review it myself, but I expect it is even better.

[I've thrown up a gist](https://gist.github.com/FeignInt/3d0cf7a867cf8884f1211837bd7a6ce0) with a suggested *mainFunction()* replacement using *case* and *glob patterns* which I hope will help

Also, assuming your editor is vim 8, I would recommend [ALE](https://github.com/dense-analysis/ale) coupled with [shellcheck](https://www.shellcheck.net) *ALE* will point out 'errors' and help you improve your scripts, it is a real time saver.

Debian package *vim-ale* is available in Debian Bullseye/Sid, I don't know about other distros, but the install instructions on Ale's github are nice and clear (I'm still using the git-plugins install, despite it being available in Sid)

**Comment by es20490446e** [comment]
@FeignInt [Video-reply](https://youtu.be/6XFmirfY1wc)

**Comment by es20490446e** [comment]
As the "Alt+F4" thing is not quite important, I have just decided to remove the work around without further checkings.

**Comment by es20490446e** [comment]
[Automount disk images](https://github.com/dosbox-staging/dosbox-staging/issues/1247)
